workers/unity/Assets/Playground/Scripts/Worker/SteamClientWorkerConnector.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Improbable.Gdk.Core;
+// using Steamworks; // Uncomment this line after installing Steamworks.NET. (https://steamworks.github.io)
+using UnityEngine;
+
+namespace Playground
+{
+    /// <summary>
+    ///     The SteamClientWorkerConnector class provides a sample solution for connecting a game client published
+    ///     on Steam to a SpatialOS deployment by retrieving a Steam auth session token at runtime.
+    ///     You can also use the ClientWorkerConnector parent class to parse the Steam auth session token
+    ///     from the commandline arguments in case the Steam auth session token is not known in advance.
+    ///     You will need to install Steamworks.NET (https://steamworks.github.io) in order to use this code.
+    /// </summary>
+    public class SteamClientWorkerConnector : ClientWorkerConnector
+    {
+        protected override bool ShouldUseLocator()
+        {
+            return true;
+        }
+
+        /// <summary>
+        ///     Creates a Locator configuration where the Steam ticket field is populated through an API call.
+        /// </summary>
+        /// <remarks>
+        ///     All other configuration parameters are populated according to the logic of the parent class implementation.
+        ///     (Based on the command line arguments)
+        /// </remarks>
+        /// <param name="workerType">The type of the worker to create.</param>
+        /// <returns>The Locator connection configuration</returns>
+        protected override LocatorConfig GetLocatorConfig(string workerType)
+        {
+            var config = base.GetLocatorConfig(workerType);
+            var result = GetSteamAuthSessionTicket(out var steamTicket, out var errorMessage);
+            if (!result)
+            {
+                Debug.LogError("Error getting Steam auth session ticket: " + errorMessage);
+            }
+            else
+            {
+                config.SetSteamTicket(steamTicket);
+            }
+
+            return config;
+        }
+
+        /// <summary>
+        ///     Retrieves a Steam auth session token (referred to as Steam ticket in the SpatialOS worker SDK) for
+        ///     authenticating a Steam user trying to join a SpatialOS deployment.
+        /// </summary>
+        /// <param name="errorMessage">Steam auth session token in case true is returned.</param>
+        /// <param name="errorMessage">Reason for failing the retrieval in case false is returned.</param>
+        /// <returns>True, if a Steam auth session token was successfully retrieved.</returns>
+        private bool GetSteamAuthSessionTicket(out string steamTicket, out string errorMessage)
+        {
+            steamTicket = string.Empty;
+            errorMessage = "Method not implemented.";
+            return false;
+
+            // Uncomment this code and remove the preceding method stub after installing Steamworks.NET. (https://steamworks.github.io)
```

Comment:
could we use preprocessor macros for these?

Reply:
Nice idea, can have this kind of workflow:
- install steamworks.net
- change project config to define preprocessor token

Reply:
I'm not fully sure here. The preprocessor tag feels like magic. We are already struggling with getting everyone to install the correct .NET SDK or Unity version and this looks like something that people might get wrong and that is hard to debug. Uncommenting code feels more "idiot proof" to me here and since this is supposed to just serve as an example (almost like a tutorial but we've written the code for you), maybe commented code is fine?

Reply:
How do we expect users to find this? Are there going to be docs signposting this?

Reply:
There will be a docs page for this. The docs are currently being stalled due to legal concerns. ;)

Reply:
How do we test this commented code?

Reply:
Sorry, the way how github presents these comments confused me. I replied on the "main thread" of this PR.

Reply:
don't like the idea of having an example that is completely commented out. This also means we can not really test it..
 Would really prefer to have some kind of switch...

How generic is this example implementation? can it be reused?

Reply:
e.g. SteamWorksNet and Facepunch are two different libraries, right? we could split it out into two different worker connectors that serve as examples or a default implementation for anyone who wants to use steam